dask_expr/expr.py

```diff
@@ -1102,6 +1102,18 @@ class Assign(Elemwise):
     def _node_label_args(self):
         return [self.frame, self.key, self.value]
 
+    def _simplify_up(self, parent):
+        if isinstance(parent, Projection):
+            columns = set(parent.columns) - {self.key}
```

Comment:
**Just a thought**: If `self.key` is **not** in `parent.columns`, then perhaps the `Assign` expression should be dropped altogether? In that case, we are essentially assigning a column and then immediately dropping it.

Reply:
You don't need to worry about this now. Just noting that it would be a relatively simple optimization to add.

Reply:
I was struggling with this when putting up the PR, but I think I fixed the bug that caused my issues since then in another PR. Works like a charm now!

Reply:
Nice!